onnx/defs/nn/defs.cc

```diff
@@ -2832,4 +2832,447 @@ ONNX_OPERATOR_SET_SCHEMA(
+              // This is treated as a 2D shape [d[0] * ... * d[axis-1], d[axis] * ... * d[rank-1]]
+              // Normalization is applied to the second dimension.
+              // Output Y has same shape as X
+              // Outputs InvStdDev have shape: [d[0], ..., d[axis-1], 1, ..., 1]
+              FunctionBuilder builder(functionProto);
+              builder.Const("FloatEpsilon", ToTensor<float>(epsilon))
+                  .Add("Epsilon = Cast (FloatEpsilon)", "to", U)
+                  .Add("XShape = Shape (X)") // shape of input tensor: 1D tensor
+                  .Add("Rank = Size (XShape)") // rank of input tensor: scalar
+                  .Add("Zero1D = Constant()", "value", mktensor(0)) // [0] : 1D tensor
+                  .Add("Axis1D = Constant()", "value", mktensor(axis)) // [axis] : 1D tensor
+                  .Add("PrefixShape = Slice (XShape, Zero1D, Axis1D)") // [d[0], ..., d[axis-1]]
+                  .Add(
+                      axis >= 0 // number of axes that are reduced =
+                          ? "NumReducedAxes = Sub (Rank, Axis1D)" // [rank - axis]: 1D tensor
+                          : "NumReducedAxes = Neg (Axis1D)") // [-axis] : 1D tensor
+                  .Add(
+                      "SuffixShape = ConstantOfShape (NumReducedAxes)",
+                      "value",
+                      mktensor(1)) // [1, ..., 1] for reduced axes
+                  .Add("ReducedShape = Concat <axis = 0> (PrefixShape, SuffixShape)") // [d[0], ..., d[axis-1], 1, ..., 1]
+                  .Add("X2D = Flatten (X)", "axis", axis)
+                  .Add("XU = Cast (X2D)", "to", U);
+              builder.Add("Axes_1 = Constant()", "value", mktensor(1))
+                    .Add("Mean2D = ReduceMean (XU, Axes_1)")
+                    .Add("Exponent = Constant <value = float {2.0}>()")
+                    .Add("Square = Pow (XU, Exponent)")
+                    .Add("MeanOfSquare = ReduceMean (Square, Axes_1)");
+              builder.Add("RMS = Pow (Mean2D, Exponent)")
+                  .Add("RMSPlusEpsilon = Add (RMS, Epsilon)")
+                  .Add("StdDev = Sqrt (VarPlusEpsilon)")
+                  .Add("Normalized = Div (XU, StdDev)")
+                  .Add("NormalizedT = Cast (Normalized)", "to", T)
+                  .Add("Scale2D = Flatten <axis = 0> (Scale)")
+                  .Add("Scaled = Mul (NormalizedT, Scale2D)");
+              builder.Add("Y = Reshape (Scaled, XShape)");
+              builder.Add("InvStdDev2D = Reciprocal (StdDev)");
+              if (ctx.hasOutput(1))
+                builder.Add("Mean = Reshape (Mean2D, ReducedShape)");
+              if (ctx.hasOutput(2))
+                builder.Add("InvStdDev = Reshape (InvStdDev2D, ReducedShape)");
+
+            schema.BuildFunction(functionProto);
+            return true;
+          }));
+
+static const char* SkipLayerNormalization_ver23_doc = R"DOC(
+
```

Comment:
Can you add some documentation here?

Reply:
Can you also pointers (URLs) to the papers defining or using it as background information?